### PR TITLE
[Hexagon] Fix RPC session close by adding shutdown PackedFunc

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR adds a shutdown packed function to RPC module so we can force close the RPC session from python side.

cc @areusch @Lunderberg

### CODE REVIEW

File: src/runtime/rpc/rpc_module.cc [hunk 1]
```diff
@@ -177,6 +177,10 @@ class RPCModuleNode final : public ModuleNode {
   const char* type_key() const final { return "rpc"; }
 
   PackedFunc GetFunction(const std::string& name, const ObjectPtr<Object>& sptr_to_self) final {
+    if (name == "Shutdown") {
```
Comment (Lunderberg) [inline src/runtime/rpc/rpc_module.cc:180]: As I'm reading it, I think we should rename the exposed function to something more descriptive, such as `CloseRPCConnection`.  I know we initially named it `Shutdown` because that's what the `RPCEndpoint` function is called, but it could cause confusion at this level.  (e.g. "Shutdown" could also mean to power off the remote device.)

File: src/runtime/rpc/rpc_endpoint.cc [hunk 2]
```diff
@@ -623,6 +623,8 @@ class RPCEndpoint::EventHandler : public dmlc::Stream {
 
 RPCCode RPCEndpoint::HandleUntilReturnEvent(bool client_mode, RPCSession::FEncodeReturn setreturn) {
   RPCCode code = RPCCode::kCallFunc;
+
+  ICHECK(channel_ != nullptr) << "channel is already closed";
```
Comment (Lunderberg) [inline src/runtime/rpc/rpc_endpoint.cc:627]: Since this would be the first error message seen by a user/developer who runs into it, we should include more context.  Also, since it could result from usage rather than being an internal error, we should use `CHECK` instead of `ICHECK`.

```c++
CHECK(channel_) << "Expected connection to server " << name_ << " to be active, but the connection was previously closed";
```

File: src/runtime/rpc/rpc_endpoint.h [hunk 3]
```diff
@@ -70,6 +70,12 @@ class RPCEndpoint {
  public:
   /*! \brief virtual destructor */
   ~RPCEndpoint();
+
+  /*!
+   *  \brief The server shutdown cleanup function.
+   */
+  void Shutdown();
```
Comment (Lunderberg) [inline src/runtime/rpc/rpc_endpoint.h:77]: We should have a more descriptive docstring here, and in the destructor.  For the destructor, we should say that it closes the connection, if the connection hasn't already been closed.  For the Shutdown function, we should 

* Shutdown has no effect if the connection has already been shut down.
* Shutdown will wait for all output currently queued from the RPC connection (i.e. The user doesn't need to wait for completion before calling Shutdown.)
* Any further use of objects that depended on the endpoint (e.g. A `tvm.nd.array` allocated on the remote RPC session) may throw an exception when used.

File: tests/scripts/task_python_hexagon.sh [hunk 4]
```diff
@@ -51,7 +51,7 @@ export ANDROID_SERIAL_NUMBER=${device_serial}
 if [ "${device_serial}" == "simulator" ]; then
     run_pytest ctypes python-contrib-hexagon tests/python/contrib/test_hexagon
 else
-    run_pytest ctypes python-contrib-hexagon tests/python/contrib/test_hexagon --tx $num_of_devices*popen --dist=load
+    run_pytest ctypes python-contrib-hexagon tests/python/contrib/test_hexagon -n=$num_of_devices
```
Comment (Lunderberg) [inline tests/scripts/task_python_hexagon.sh:54]: This looks like an unrelated change.  Should it be part of this PR?

File: src/runtime/rpc/rpc_session.h [hunk 5]
```diff
@@ -253,6 +253,8 @@ class RPCSession {
    */
   static std::shared_ptr<RPCSession> Get(int table_index);
 
+  virtual void Shutdown() {}
```
Comment (Lunderberg) [inline src/runtime/rpc/rpc_session.h:256]: Docstring needed

**Review by Lunderberg — CHANGES_REQUESTED** [review]
Overall, looks good, just a couple of changes to naming/documentation.

File: src/runtime/rpc/rpc_session.h (hunk 5 above)
Comment (mehrdadh) [reply src/runtime/rpc/rpc_session.h:259]: added

**Review by Lunderberg — APPROVED** [review]
Thank you for the changes, and looks good!

**Comment by mehrdadh** [comment]
fixes https://github.com/apache/tvm/issues/12779
